src/java.desktop/unix/native/libawt_xawt/xawt/XToolkit.c

```diff
@@ -692,7 +692,9 @@ void awt_output_flush() {
 static void wakeUp() {
     static char wakeUp_char = 'p';
     if (!isMainThread() && awt_pipe_inited) {
-        write ( AWT_WRITEPIPE, &wakeUp_char, 1 );
+        if (write(AWT_WRITEPIPE, &wakeUp_char, 1) < 0) {
```

Comment:
No strong preference between write() < 0, < 1, or != 1, it depends on how strict we want to be.